main.go

```diff
@@ -132,16 +216,31 @@ func caduceus(arguments []string) int {
 		modifiedWRPCount:         metricsRegistry.NewCounter(ModifiedWRPCounter),
 		maxOutstanding:           0,
 	}
+
 	caduceusConfig.Webhook.Logger = logger
 	caduceusConfig.Webhook.MetricsProvider = metricsRegistry
-	svc, stopWatches, err := ancla.Initialize(caduceusConfig.Webhook, caduceusSenderWrapper)
+	argusClientTimeout, err := newArgusClientTimeout(v)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to parse argus client timeout config values: %s \n", err.Error())
+		return 1
+	}
 
+	caduceusConfig.Webhook.Argus.HTTPClient = newHTTPClient(argusClientTimeout, tracing)
+	svc, stopWatches, err := ancla.Initialize(caduceusConfig.Webhook, GetLogger, caduceusSenderWrapper)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Webhook service initialization error: %v\n", err)
 		return 1
 	}
+	infoLogger.Log(logging.MessageKey(), "Webhook service enabled")
+
+	rootRouter := mux.NewRouter()
+	otelMuxOptions := []otelmux.Option{
+		otelmux.WithPropagators(tracing.Propagator),
+		otelmux.WithTracerProvider(tracing.TracerProvider),
+	}
+	rootRouter.Use(otelmux.Middleware("mainSpan", otelMuxOptions...), candlelight.EchoFirstTraceNodeInfo(tracing.Propagator))
```

Comment:
@joe94 Is `mainSpan` the name we want for this server? I'm not sure the repercussions of this, but I would think either `caduceus` or `primary`?

Reply:
Or maybe take the value from `candelight.Config.ApplicationName`?
https://pkg.go.dev/github.com/xmidt-org/candlelight#Config

Reply:
I think we have released all the other microservices with the same name. So we can keep it like this for now.

Reply:
Yeah, I called it `mainSpan` in the earlier services we integrated because we were only instrumenting one server. I just saw, though, that in Talaria we have both the `primary` and the `control` servers so I think it'll be good to use the server names from now on. 

@utsavbatra5, let's rename this to `primary`. We can retroactively update the other services as we go. 

From what I saw using the Zipkin UI, the node name (which comes from `candlelight.Config.ApplicationName`) is the most important visual cue in the trace graph. This operation name will mostly help when inspecting specific requests.

Reply:
I have fixed the same in latest commit.